src/components/modals/ModalComponent.tsx

```diff
@@ -69,6 +70,9 @@ class ModalComponent extends React.Component<ModalProps> {
   close = () => {
     this.animateAndClose!()
     this.props.onClose?.()
+    if (Capacitor.isNativePlatform()) {
+      window.location.reload()
+    }
```

Comment:
This is good that it only executes on the Capacitor build, but let's also avoid refreshing when the theme has not changed. As mentioned previously, we only want to refresh the app if we absolutely have to, since it is highly undesirable from a UX perspective. Hope you can understand.

You should have an easier time accessing Settings-specific logic by tapping into the `onClose` hook in [ModalSettings](https://github.com/cybersemics/em/blob/e39ebc414b562bd1b6a8131cdea6977da51bd273/src/components/modals/Settings.tsx#L117). Actually, this needs to be moved there anyway, because as-written it will refresh the app whenever any modal is closed.

Reply:
Gotcha, I added a workaround to make it refresh as minimally as possible. It will only work onClose for the settings modal menu in this new version.